plugins/poll/app/controllers/polls_controller.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+class DiscoursePoll::PollsController < ::ApplicationController
+  requires_plugin DiscoursePoll::PLUGIN_NAME
+
+  before_action :ensure_logged_in, except: [:voters, :grouped_poll_results]
+
+  def vote
+    post_id = params.require(:post_id)
+    poll_name = params.require(:poll_name)
+    options = params.require(:options)
+
+    begin
+      poll, options = DiscoursePoll::Poll.vote(post_id, poll_name, options, current_user)
```

Comment:
Love that you've made a method for each operations 👏 

I would have put the `current_user` as the 1st parameter but that's nitpicking.